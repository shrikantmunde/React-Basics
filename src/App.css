.app {
  background-color: black;
  color: red;
}
