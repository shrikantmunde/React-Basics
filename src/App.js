import React from 'react';

function App(props) {
  const style = {
    color: '#ff0000',
    backgroundColor: 'yellow'
  }

  if (props.border == "Red") {
    style.color = 'Black',
    style.border = '2px solid blue'
    style.backgroundColor = 'skyblue'
  }

  return (
    <div style={style}>
      <h2>{props.title}</h2>
      <p>{props.subtitle2}</p>
      <p>Another line</p>
      {props.children}
    </div>
  );
}

export default App;
