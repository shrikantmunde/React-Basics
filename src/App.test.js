import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

it('renders with a title', function() {
  const component = renderer.create(<App title="My Title" />);
  expect(component).toMatchSnapshot();
})

it('renders with a border', function() {
  const component = renderer.create(<App title1="My Title" border />);
  expect(component).toMatchSnapshot();
})
