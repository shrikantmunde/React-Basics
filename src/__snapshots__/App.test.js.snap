exports[`test renders with a border 1`] = `
<div
  style={
    Object {
      "backgroundColor": "yellow",
      "color": "#ff0000",
    }
  }>
  <h2 />
  <p />
  <p>
    Another line
  </p>
</div>
`;

exports[`test renders with a title 1`] = `
<div
  style={
    Object {
      "backgroundColor": "yellow",
      "color": "#ff0000",
    }
  }>
  <h2>
    My Title
  </h2>
  <p />
  <p>
    Another line
  </p>
</div>
`;
